elastic/kibana
# [SecuritySolution] Reveal data quality entry on the side-nav in serverless

## Summary

Test env 
serverless: https://p.elstc.co/paste/4JjGBmXv#Yogkkphe2hFyBRWIhsAZ3ivsvAve4TJkS2BzOqPQp3w
~ESS: https://p.elstc.co/paste/EsCQBV7V#P18h4HGLZkrPm66+KB2WVH6srVwDwXqVsciQDtI8vbc~


<img width="2559" alt="Screenshot 2024-04-17 at 13 48 08" src="https://github.com/elastic/kibana/assets/6295984/558075d9-f58a-458e-8006-7506ab432f5a">
<img width="2558" alt="Screenshot 2024-04-17 at 13 48 26" src="https://github.com/elastic/kibana/assets/6295984/7010e3c0-e0bb-4981-8a09-f478ed0fe018">

https://github.com/elastic/kibana/assets/6295984/5776c9c2-2957-480e-8a2c-fdd687cdebc5




We can merge this PR once all the dependencies are merged.

dependencies:
1. https://github.com/elastic/kibana/pull/179666
2. https://github.com/elastic/elasticsearch-serverless/pull/1482
3. ~https://github.com/elastic/elasticsearch-serverless/pull/1705~
4. https://github.com/elastic/elasticsearch-serverless/pull/1753

Steps to verify:

0. Enable Docker
1. Start serverless ES with the commit where the stats api was merged:
```
yarn es serverless --projectType security --tag git-2eb724699bc0
```

start Kibana
```
yarn serverless-security --no-base-path
```

2. ~Endable the settings~ This step in no longer needed.
```
curl -u {YOUR_ELASTIC_USERNAME}:{ELASTIC_PASSWORD} -X PUT "localhost:9200/_cluster/settings?pretty" -H 'Content-Type: application/json' -d'
{
  "persistent" : {
    "metering.index-info-task.enabled" : true
  }
}
```
3. In Kibana dev tool, add some testing data:
```
PUT my-test-index

PUT my-test-index/_mapping
{
"properties": {
  "@timestamp": {
    "type": "date"
  },
  "agent.type": {
    "type": "constant_keyword"
  },
  "event.category": {
    "type": "constant_keyword"
  }
}
}

POST my-test-index/_doc
{
"@timestamp": "2024-04-22T09:41:49.668Z",
"host": {
  "name": "foo"
},
"event": {
  "category": "an_invalid_category"
},
"some.field": "this",
"source": {
  "port": 90210,
  "ip": "10.1.2.3"
}
}

POST my-test-index/_doc
{
"@timestamp": "2024-04-22T09:42:22.123Z",
"host": {
  "name": "bar"
},
"event": {
  "category": "an_invalid_category"
},
"some.field": "space",
"source": {
  "port": 867,
  "ip": "10.9.8.7"
}
}
```

5. Go to /app/management/kibana/dataViews/dataView/security-solution-default , and add `my-test-*` to default index pattern
6. Go to `/app/security/data_quality` and verify that there's no doc size in the dashboard. Dashboard and treemap render correctly.
